### PR TITLE
DE Salt Browser Tests 

### PR DESCRIPTION
**Pull request by dalejerikson**
## Description
ACAS Client Routes and Tests for the Newly Implemented Salt Browser 

(My bad. Forgot to do PR for this so adding it later than I should.) 

Modifies Route For: 
* Create Salt - Adds Dryrun Param 
* Get Salts - Changes Backend Route Call for Consistency Purposes 

Adds Routes For:
* Search 
* Get All Salts in SDF Format 
* Delete Salt
* Edit Salt (Dryrun and Non-Dryrun) 
* Render Salt Structure 

Tests the common use cases and edge cases of each of the above 

## How Has This Been Tested?
TDD

### CODE REVIEW

**Review by bffrost — CHANGES_REQUESTED** [review]

**Comment by dalejerikson** [comment]
Friendly reminder to look at PR in case notification slipped past ya'll @bffrost @hhan-schrodinger :)

File: tests/test_acasclient.py [hunk 1]
```diff
@@ -2352,6 +2353,191 @@ def test_50_delete_ls_thing(self):
...
+        dryrun = "false"
+        abbrev = "I"
+        name = "Iodine"
+        mol_structure = "\n  Ketcher 08172213312D 1   1.00000     0.00000     0\n\n  1  0  0     1  0            999 V2000\n    7.4500   -7.1750    0.0000 I   0  0  0  0  0  0  0  0  0  0  0  0\nM  CHG  1   1  -1\nM  END\n"
+        self.client.create_salt(abbrev, name, mol_structure, dryrun)
+        # Try to Change Benzoic Acid Abbreviation and Name to Iodine's Abbreviation and Name 
+        search_result = self.client.search_salts("Benzoic Acid")
+        id = search_result[0]["id"]
+        updated_struct = search_result[0]["molStructure"]
+        self.assertIn("INDIGO", updated_struct)
```
Comment (bffrost) [inline tests/test_acasclient.py:2482]: nit: This step might fail if we're using a different chemistry engine. Maybe just check that the molStructure is non-empty?

**Review by bffrost — CHANGES_REQUESTED** [review]
Thanks for adding these tests @dalejerikson . Left some nits and some requests for a bit more explanation, but this looks very close to ready.

File: acasclient/acasclient.py [hunk 2]
```diff
@@ -1669,23 +1669,63 @@ def delete_stereo_category(self, id: int) -> bool:
...
+
+    def edit_salt(self, id, dryrun, abbrev, name, mol_structure):
+        """
+        Edit a salt
+        """
+        resp = self.session.put("{}/api/cmpdRegAdmin/salts/edit/{}?dryrun={}".format(self.url, id, dryrun), json={"abbrev": abbrev, "name": name, "molStructure": mol_structure})
+        resp.raise_for_status()
+        return resp.json()
+
+    def render_salt(self, mol_structure, height, width, format):
```
[the remark this answers is not in the document]
Comment (dalejerikson) [reply acasclient/acasclient.py:1721]: I was mentioned in a meeting a while ago by either you or Brian Bolt to add this for testing consistency but I can remove it

File: tests/test_acasclient.py [hunk 3]
```diff
@@ -2352,6 +2353,191 @@ def test_50_delete_ls_thing(self):
...
+
+    def test_52_get_all_salts_no_data(self):
+        self.removeAllSalts()
+        salts = self.client.get_salts()
+        self.assertEqual([], salts)
+        
+    def test_53_get_all_salts_sdf(self):
+        self.registerTestSalt()
+        salts_sdf = self.client.get_salts_sdf()
+        self.assertNotEqual([], salts_sdf)
```
[the remark this answers is not in the document]
Comment (dalejerikson) [reply tests/test_acasclient.py:2382]: Yep!

File: tests/test_acasclient.py [hunk 4]
```diff
@@ -2352,6 +2353,191 @@ def test_50_delete_ls_thing(self):
                                             code, None)
         self.assertIn(True, [ls_thing['deleted']])
 
+    def registerTestSalt(self):
```
[the remark this answers is not in the document]
Comment (dalejerikson) [reply tests/test_acasclient.py:2356]: Will do!

**Comment by bffrost** [comment]
@dalejerikson can you check up on this older PR and get this merged please? Don't want to forget about it!

File: tests/test_acasclient.py [hunk 5]
```diff
@@ -2352,6 +2353,191 @@ def test_50_delete_ls_thing(self):
...
+        updated_name = "Ethyl benzoate"
+        updated_abbrev = "BzCOOCH2CH3"
+        dryrun = "false"
+        edit_response = self.client.edit_salt(id, dryrun, updated_abbrev, updated_name, updated_struct)
```
Comment (hhan-schrodinger) [inline tests/test_acasclient.py:2533]: I'm wondering if you need this line if edit_response isn't checked like search_result/delete_response? I'm not too familiar with this so more curious than anything.

**Review by hhan-schrodinger — APPROVED** [review]
Looks good to me! I found one test with a line that might be redundant or missing an assert function? This is a really helpful example for me writing acasclient tests for the endpoint manager! 👍

**Comment by dalejerikson** [comment]
(Didn't forget this. Seems like there's memory / docker disk space issue on my old computer as I'm experiencing problems with another acascleint PR. Will push/rebase my changes and rerequest review after my new computer migration and other PR is merged)
